common/djangoapps/student/views.py

```diff
@@ -1384,7 +1384,9 @@ def login_user(request, error=""):  # pylint: disable=too-many-statements,unused
             user = User.objects.get(email=email)
         except User.DoesNotExist:
             try:
-                user = User.objects.get(username=email)
+                # try using the email as username
+                username = email
```

Comment:
Yeah, I get what you're trying to do. But for a feature as significant as this (which also presents potential security risks by its very nature), it'd be good to consider a more complete solution. I don't know what that would be or else I'd share it, but the way it currently is it feels like it could come and bite down the road.

If you still would like to move forward with this technique you can but the edX review would probably come back with a similar issue.

P.S. I think we're making changes for the old view anyway. Shouldn't you be doing this in `student_account.views`?